.github/workflows/gradle-cache-check.yml

```diff
@@ -0,0 +1,31 @@
+name: Periodic Gradle Build Cache Check
+on:
+  schedule:
+    - cron: '0 01 * * 1'
+
+env:
+  LC_ALL: "en_US.UTF-8"
+  GRADLE_ENTERPRISE_ACCESS_KEY: ${{ secrets.GRADLE_ENTERPRISE_ACCESS_KEY }}
+
+jobs:
+  build-cache-check:
+    if: github.repository == 'line/armeria'
+    runs-on: self-hosted
+    steps:
+      - id: setup-jdk-19
+        name: Set up JDK 19
+        uses: actions/setup-java@v3
+        with:
+          distribution: 'temurin'
+          java-version: '19'
+
+      - uses: gradle/gradle-enterprise-build-validation-scripts/.github/actions/gradle/download@actions-stable
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
+
+      - uses: gradle/gradle-enterprise-build-validation-scripts/.github/actions/gradle/experiment-3@actions-stable
```

Comment:
As @runningcode pointed out in a comment, we'd also suggest also running [experiment 1](https://github.com/gradle/gradle-enterprise-build-validation-scripts/blob/main/.github/actions/gradle/experiment-1/action.yml) to verify incremental builds.